src/test/operators/table_scan_sorted_segment_search_test.cpp

```diff
@@ -0,0 +1,106 @@
+#include "base_test.hpp"
+#include "gtest/gtest.h"
+#include "operators/table_scan/sorted_segment_search.hpp"
+#include "storage/segment_iterate.hpp"
+
+namespace opossum {
+
+using TestData = std::tuple<std::string, PredicateCondition, int32_t, std::vector<int32_t>>;
+using Params = std::tuple<TestData, OrderByMode, bool>;
+
+namespace {
+
+auto formatter = [](const ::testing::TestParamInfo<Params> info) {
+  return order_by_mode_to_string.at(std::get<1>(info.param)) + std::get<0>(std::get<0>(info.param)) +
+         (std::get<2>(info.param) ? "WithNulls" : "WithoutNulls");
+};
+
+}  // namespace
+
+class OperatorsTableScanSortedSegmentSearchTest : public BaseTest, public ::testing::WithParamInterface<Params> {
+ protected:
+  void SetUp() override {
+    bool nullable;
+    TestData test_data;
+    std::tie(test_data, _order_by, nullable) = GetParam();
+    std::tie(std::ignore, _predicate_condition, _search_value, _expected) = test_data;
+
+    const bool ascending = _order_by == OrderByMode::Ascending || _order_by == OrderByMode::AscendingNullsLast;
+    const bool nulls_first = _order_by == OrderByMode::Ascending || _order_by == OrderByMode::Descending;
+
+    if (!ascending) {
+      std::reverse(_expected.begin(), _expected.end());
+    }
+
+    _segment = std::make_unique<ValueSegment<int32_t>>(nullable);
+
+    if (nullable && nulls_first) {
+      _segment->append(NULL_VALUE);
+      _segment->append(NULL_VALUE);
+      _segment->append(NULL_VALUE);
+    }
+
+    const auto table_size = 10;
+    for (int32_t row = 0; row < table_size; ++row) {
```

Comment:
"ChunkOffset", not "int32_t"

Reply:
I don't understand why we should use `ChunkOffset` here because we are not accessing a segment via this "offset". It's just to add ints to our table. Can you please elaborate?

Reply:
Sorry, I misread the loop. "row" is often the name for a offset within a Chunk (then using ChunkOffset as a type would be correct). My bad.